var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -39,6 +39,16 @@ class M4(Package):
 
     depends_on('libsigsegv', when='+sigsegv')
 
+    def _make_executable(self, name):
+        return Executable(join_path(self.prefix.bin, name))
+
+    def setup_dependent_package(self, module, dependent_spec):
+        # m4 is very likely to be a build dependency, so we add the
```

Comment:
Is the build environment not adding `m4` to the path even when it is a run dep of `autoconf`?  If it's not, it probably should, or we're setting the environment up wrong by default.